nucliadb_vectors/src/data_point/ops_hnsw.rs

```diff
@@ -27,41 +27,38 @@ use rand::prelude::*;
 use rand::rngs::SmallRng;
 
 use super::*;
+use crate::data_point::params;
 
-pub mod params {
-    pub fn level_factor() -> f64 {
-        1.0 / (m() as f64).ln()
-    }
-    pub const fn m_max() -> usize {
-        30
-    }
-    pub const fn m() -> usize {
-        30
-    }
-    pub const fn ef_construction() -> usize {
-        100
-    }
-}
+/// Implementors of this trait can guide the hnsw search
 pub trait DataRetriever: std::marker::Sync {
-    fn get_key(&self, _: Address) -> &[u8];
-    fn is_deleted(&self, _: Address) -> bool;
-    fn has_label(&self, _: Address, _: &[u8]) -> bool;
-    fn similarity(&self, _: Address, _: Address) -> f32;
-    fn get_vector(&self, _: Address) -> &[u8];
+    /// Returns the key of the embedding pointed by x.
+    fn get_key(&self, x: Address) -> &[u8];
+    /// Checks if x is a deleted embedding.
+    fn is_deleted(&self, x: Address) -> bool;
+    /// During filtering this function will be called to know if x has the necessary labels.
+    fn has_label(&self, x: Address, label: &[u8]) -> bool;
+    /// Computes the similarity between the embeddings pointed by x and y.
+    fn similarity(&self, x: Address, y: Address) -> f32;
+    /// Returns the serialized embedding of x, is meant to be used by [`crate::data_types::vector`].
+    fn get_vector(&self, x: Address) -> &[u8];
```

Comment:
I don't think functions with good names as, for example, `is_deleted(...) -> bool` need more explanation. The only difference between reading the function signature and the documentation is that you're calling an `Address` and embedding. If that's a crucial part, consider renaming the type.

Reply:
An `Address` is a pointer to an embedding, will document that part of the code in the next round.